extensions/queries/QueriesController.php

```diff
@@ -408,7 +408,11 @@ public function savequeryAction()
         $res = "json or desc missing";
         // checking for post data to save queries
         $params = $this->_request->getParams();
-        if (isset($params['json']) && isset($params['json'])) {
+        $hasAC = !(
+            empty(Erfurt_App::getInstance()->isActionAllowed('ModelManagement'))
```

Comment:
What is the call to `empty` good for? The `isActionAllowed` returns a bool.

Please also consider to check whether the current model is writable, for the "share with others" case (not sure whether this is currently checked somewhere, please verify).